# <u>**ACADEMY**</u> : Educational Systems Data Analysis <a id='top'></a>

<details open>
<summary><u style="font-size: 2rem">Summary: </u></summary>

* <a href="#author" data-toc-modified-id="author" >Author</a>
* <a href="#introduction" data-toc-modified-id="introduction" >Introduction</a>
* <a href="#functions" data-toc-modified-id="functions" >Functions</a>
* <a href="#import-necessary-libraries" data-toc-modified-id="import-necessary-libraries" >Import Necessary Libraries</a>
* <a href="#load-the-data" data-toc-modified-id="load-the-data" >Load the Data</a>
* <a href="#initial-data-exploration" data-toc-modified-id="initial-data-exploration">Initial Data Exploration</a>
    * <a href="#stats-data" data-toc-modified-id="stats-data">1 - EdStatsData</a>
    * <a href="#stats-country" data-toc-modified-id="stats-country">2 - EdStatsCountry</a>
    * <a href="#stats-country-series" data-toc-modified-id="stats-country-series">3 - EdStatsCountry-Series</a>
    * <a href="#stats-series" data-toc-modified-id="stats-series">4 - EdStatsSeries</a>
    * <a href="#stats-foot-note" data-toc-modified-id="stats-foot-note">5 - EdStatsFootNote</a>
    * <a href="#analysis-note" data-toc-modified-id="analysis-note"><u>Analysis note</u></a>
* <a href="#data-cleaning" data-toc-modified-id="data-cleaning">Data Cleaning</a>
    * <a href="#relevant indicators" data-toc-modified-id="relevant indicators">Relevant Indicators</a>
    * <a href="#filter-&-eliminate-unnecessary-columns" data-toc-modified-id="filter-&-eliminate-unnecessary-columns">Filter & Eliminate Unnecessary Columns</a>
        * <a href="#cleaning-stats-data" data-toc-modified-id="cleaning-stats-data">1 - EdStatsData</a>
        * <a href="#cleaning-stats-data" data-toc-modified-id="cleaning-stats-data">2 - EdStatsCountry</a>
        * <a href="#cleaning-stats-country-series" data-toc-modified-id="cleaning-stats-country-series">3 - EdStatsCountry-Series</a>
        * <a href="#cleaning-series" data-toc-modified-id="cleaning-series">4 - EdStatsSeries</a>
        * <a href="#cleaning-stats-foot-note" data-toc-modified-id="cleaning-stats-foot-note">5 - EdStatsFootNote</a>


</details>

## Author <a id='author'></a>
> Mohamed Ali EL HAMECH 

- @MasterCodeDevelop (Github)
- E-mail : master.code.develop@gmail.com
- Affiliation : ACADEMY #OpenClassRooms
<p><img align="left" src="https://github-readme-stats.vercel.app/api/top-langs?username=mastercodedevelop&show_icons=true&locale=en&layout=compact" alt="mastercodedevelop" />

## Introduction:<a id='introduction'></a>
>In the rapidly evolving landscape of online education, ACADEMY, a burgeoning EdTech start-up, has been making significant strides by offering high-quality online training content tailored for high school and university students. As part of its strategic vision, ACADEMY is keenly exploring opportunities for international expansion, aiming to tap into markets with a robust educational framework and a potential clientele for its services.

_The objective of this analysis is twofold. Firstly, it seeks to delve into global educational data, sourced from the World Bank, to ascertain the viability and potential of various countries as prospective markets for ACADEMY. This dataset, curated by the "EdStats All Indicator Query" of the World Bank, boasts a comprehensive collection of over 4,000 international indicators. These indicators span a range of metrics, from access to education and graduation rates to insights about educators and educational expenditures._

_Secondly, this analysis aims to provide a clear, data-driven narrative that would aid ACADEMY's decision-makers in charting the company's international trajectory. By evaluating the quality of the dataset, understanding its breadth and depth, and extracting relevant insights, we hope to offer a roadmap that aligns with ACADEMY's mission and vision._

## Import Necessary Libraries <a id='import-necessary-libraries'></a>
> Before you can work with the data, you need to import the necessary libraries.

In [177]:
# Pandas is used for data manipulation and analysis
import pandas as pd

# Numpy is used for mathematical and logical operations on arrays
import numpy as np 

[Back to top](#top)

## Functions <a id='functions'></a>

In [178]:
def title(text):
    """
    Format the given text with an underline.
    
    Parameters:
    - text (str): The input string to be formatted.
    
    Returns:
    - str: The formatted string with an underline with a line break.
    
    Example:
    --------
    >>> title("Hello World")
    '\x1B[4mHello World\x1B[0m \n'
    """
    
    # ANSI escape code '\x1B[4m' is used to start the underline effect
    # ANSI escape code '\x1B[0m' is used to reset the formatting
    return f"\x1B[4m" + text + "\x1B[0m \n"

In [179]:
def missing_values_table(df, percentage = 0):
    """
    Return a DataFrame displaying the count and percentage of missing values for each column.
    
    Parameters:
    - df (DataFrame): The input DataFrame to check for missing values.
    
    Returns:
    - DataFrame: A DataFrame with columns 'Missing Values' and 'Missing Percentages', sorted in descending order of missing percentages.
    """
    
    # Calculate missing values and their percentages
    missing_values = df.isnull().sum()
    missing_percentages = (df.isnull().mean() * 100).round(2)  # rounding to 2 decimal places for clarity
    
    # Create a DataFrame to display missing data info
    missing_data = pd.DataFrame({
        'Missing Values': missing_values, 
        'Missing Percentages (%)': missing_percentages
    })
    
    # Filtering columns based on the given percentage and sorting by percentage
    missing_data = missing_data[(missing_data['Missing Values'] > 0) & (missing_data['Missing Percentages (%)'] >= percentage)]
    missing_data = missing_data.sort_values(by='Missing Percentages (%)', ascending=False)
    
    # Display if percentage is greater than 0
    if percentage > 0:
        print(title(f"Displaying columns with missing data >= {percentage}%:"))
        print(f"Total columns that have missing data >= {percentage}%: {missing_data.index.size}")
    else:
        print(title("Displaying all columns with missing data:"))
    
    if missing_data.empty:
        print("There are no columns with the specified percentage of missing data.")
    else:
        return missing_data

In [180]:
def display_duplicate_rows(df):
    """
    Display duplicate rows of a DataFrame.
    
    Parameters:
    - df (DataFrame): The input DataFrame to check for duplicate rows.
    
    Returns:
    - None: Prints the number of duplicate rows or a message indicating no duplicates.
    """
    
    # Find duplicate rows
    duplicated_rows = df[df.duplicated()]
    duplicated_rows_sum = duplicated_rows.shape[0]

    if duplicated_rows_sum > 0:
        print(title("Total duplicate rows:"+ duplicated_rows_sum))
        print("Displaying duplicate rows:")
        display(duplicated_rows)  # This will display the DataFrame in a Jupyter Notebook environment
    else:
        print("There are no duplicate rows.")

In [181]:
def drop_columns(df, percentage=100):
    """
    Drop columns from a DataFrame based on a specified missing value percentage threshold.
    
    Parameters:
    - df (DataFrame): The input DataFrame from which columns will be dropped.
    - percentage (float, optional): The missing value percentage threshold. Columns with missing values 
      equal to or greater than this threshold will be dropped. Default is 100.
    
    Returns:
    - None: The function modifies the DataFrame in-place and prints the result of the operation.
    
    Example:
    --------
    >>> df = pd.DataFrame({'A': [1, 2, np.nan], 'B': [np.nan, np.nan, np.nan]})
    >>> drop_columns(df)
    1 column dropped because it had 100% missing values: B
    """
    
    # Get the missing data information using the previously defined function
    missing_data = missing_values_table(df)
    
    # Identify columns to drop based on the given percentage threshold
    columns_to_drop = missing_data[missing_data['Missing Percentages (%)'] >= percentage].index.tolist()
    
    # Drop the identified columns and print the result
    if columns_to_drop:
        df.drop(columns_to_drop, axis=1, inplace=True)
        plural = 's' if len(columns_to_drop) > 1 else ''
        print(f"{len(columns_to_drop)} column{plural} dropped because it{plural if len(columns_to_drop) == 1 else ' they'} had {percentage}% missing values: {', '.join(columns_to_drop)}")
    else:
        print(f"No columns were dropped as none had {percentage}% missing values.")

In [182]:
def explore_df(df, df_name):
    """
    Explores a given DataFrame by displaying key information.
    
    Paramètres:
    - df (DataFrame) : DataFrame to explore.
    - df_name (str) : DataFrame name for display.
    """
    print(title(f"Exploration of {df_name} \n\n"))
    
    # 1. Preview the first and last lines
    print(title(f"1. Preview the first and last line of {df_name}"))
    display(df.head())
    display(df.tail())
    
    # 2. General informations
    print(title(f"2. General information about {df_name}"))
    df.info()
    
    # 3. Data types
    print(title(f"3. Data types in {df_name}"))
    print(df.dtypes.value_counts())
    
    # 4. Valeurs manquantes
    print(title(f"4. Missing values in {df_name}"))
    display(missing_values_table(df))
    
    # 5. Missing values
    print(title(f"5. Duplicate rows in {df_name}"))
    display_duplicate_rows(df)
    
    # 6. Descriptive statistics
    print(title(f"6. Descriptive statistics for {df_name}"))
    display(df.describe())
    
    # 7. Unique values for categorical columns
    print(title(f"7. Unique values for categorical columns of {df_name}"))
    for column in df.select_dtypes(include=['object']).columns:
        print(f"Unique values for {column}: {df[column].nunique()}")
        print(df[column].unique())
        print("\n")
    
    # 8. DataFrame Dimensions
    print(title(f"\n 8. Dimensions of {df_name}"))
    print(f"Number of lines: {df.shape[0]}")
    print(f"Number of columns: {df.shape[1]}")

[Back to top](#top)

## Load the Data <a id='load-the-data'></a>
> Importing the dataset into the environment for analysis and exploration.

In [183]:
# Load the original datasetsCOUNTRY_SERIES = pd.read_csv('./data/EdStatsCountry-Series.csv')
COUNTRY_SERIES = pd.read_csv('./data/EdStatsCountry-Series.csv')
COUNTRY = pd.read_csv('./data/EdStatsCountry.csv')
DATA = pd.read_csv('./data/EdStatsData.csv')
FOOT_NOTE = pd.read_csv('./data/EdStatsFootNote.csv')
SERIES = pd.read_csv('./data/EdStatsSeries.csv')

# Create copies of the original datasets for analysis
country_series = COUNTRY_SERIES.copy()
country = COUNTRY.copy()
data = DATA.copy()
foot_note = FOOT_NOTE.copy()
series = SERIES.copy()

[Back to top](#top)

## Initial Data Exploration <a id='initial-data-exploration'></a>
> Diving into the dataset to understand its structure, content, and characteristics.

### 1 - <u>EdStatsData :</u><a id='stats-data'></a>

In [184]:
# Exploring EdStatsData
explore_df(DATA, "EdStatsData")

Exploration of EdStatsData 

 

1. Preview the first and last line of EdStatsData 



Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   
2   Arab World          ARB   
3   Arab World          ARB   
4   Arab World          ARB   

                                      Indicator Name  Indicator Code  \
0  Adjusted net enrolment rate, lower secondary, ...      UIS.NERA.2   
1  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.F   
2  Adjusted net enrolment rate, lower secondary, ...  UIS.NERA.2.GPI   
3  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.M   
4  Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   

        1970       1971       1972       1973       1974      1975  ...  2060  \
0        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
1        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
2        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
3        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
4  54.822121  54.894138  56.209438  57.267109  57.991138  59.36554  ...   NaN   

   2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[5 rows x 70 columns]

Country Name Country Code  \
886925     Zimbabwe          ZWE   
886926     Zimbabwe          ZWE   
886927     Zimbabwe          ZWE   
886928     Zimbabwe          ZWE   
886929     Zimbabwe          ZWE   

                                           Indicator Name  \
886925  Youth illiterate population, 15-24 years, male...   
886926  Youth literacy rate, population 15-24 years, b...   
886927  Youth literacy rate, population 15-24 years, f...   
886928  Youth literacy rate, population 15-24 years, g...   
886929  Youth literacy rate, population 15-24 years, m...   

              Indicator Code  1970  1971  1972  1973  1974  1975  ...  2060  \
886925      UIS.LP.AG15T24.M   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
886926     SE.ADT.1524.LT.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
886927  SE.ADT.1524.LT.FE.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
886928  SE.ADT.1524.LT.FM.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
886929  SE.ADT.1524.LT.MA.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   

        2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
886925   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
886926   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
886927   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
886928   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
886929   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[5 rows x 70 columns]

2. General information about EdStatsData 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 886930 entries, 0 to 886929
Data columns (total 70 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    886930 non-null  object 
 1   Country Code    886930 non-null  object 
 2   Indicator Name  886930 non-null  object 
 3   Indicator Code  886930 non-null  object 
 4   1970            72288 non-null   float64
 5   1971            35537 non-null   float64
 6   1972            35619 non-null   float64
 7   1973            35545 non-null   float64
 8   1974            35730 non-null   float64
 9   1975            87306 non-null   float64
 10  1976            37483 non-null   float64
 11  1977            37574 non-null   float64
 12  1978            37576 non-null   float64
 13  1979            36809 non-null   float64
 14  1980            89122 non-null   float64
 15  1981            38777 non-null   float64
 16  1982         

Missing Values  Missing Percentages (%)
Unnamed: 69          886930                   100.00
2017                 886787                    99.98
2016                 870470                    98.14
1971                 851393                    95.99
1973                 851385                    95.99
...                     ...                      ...
2011                 740918                    83.54
2012                 739666                    83.40
2000                 710254                    80.08
2005                 702822                    79.24
2010                 644488                    72.67

[66 rows x 2 columns]

5. Duplicate rows in EdStatsData 

There are no duplicate rows.
6. Descriptive statistics for EdStatsData 



1970          1971          1972          1973          1974  \
count  7.228800e+04  3.553700e+04  3.561900e+04  3.554500e+04  3.573000e+04   
mean   1.974772e+09  4.253638e+09  4.592365e+09  5.105006e+09  5.401493e+09   
std    1.211687e+11  1.804814e+11  1.914083e+11  2.059170e+11  2.112150e+11   
min   -1.435564e+00 -1.594625e+00 -3.056522e+00 -4.032582e+00 -4.213563e+00   
25%    8.900000e-01  8.853210e+00  9.240920e+00  9.595200e+00  9.861595e+00   
50%    6.317724e+00  6.316240e+01  6.655139e+01  6.969595e+01  7.087760e+01   
75%    6.251250e+01  5.655200e+04  5.863650e+04  6.202900e+04  6.383675e+04   
max    1.903929e+13  1.986457e+13  2.100916e+13  2.238367e+13  2.282991e+13   

               1975          1976          1977          1978          1979  \
count  8.730600e+04  3.748300e+04  3.757400e+04  3.757600e+04  3.680900e+04   
mean   2.314288e+09  5.731808e+09  6.124437e+09  6.671489e+09  7.436724e+09   
std    1.375059e+11  2.215546e+11  2.325489e+11  2.473986e+11  2.660957e+11   
min   -3.658569e+00 -2.950945e+00 -3.174870e+00 -3.558749e+00 -2.973612e+00   
25%    1.400000e+00  9.312615e+00  9.519913e+00  1.000000e+01  1.000000e+01   
50%    9.677420e+00  7.101590e+01  7.133326e+01  7.290512e+01  7.510173e+01   
75%    7.854163e+01  5.682800e+04  5.739175e+04  5.940425e+04  6.411500e+04   
max    2.300634e+13  2.424128e+13  2.521383e+13  2.622101e+13  2.730873e+13   

       ...          2060          2065          2070          2075  \
count  ...  5.143600e+04  5.143600e+04  5.143600e+04  5.143600e+04   
mean   ...  7.224868e+02  7.271290e+02  7.283779e+02  7.266484e+02   
std    ...  2.215845e+04  2.287990e+04  2.352338e+04  2.408149e+04   
min    ... -1.630000e+00 -1.440000e+00 -1.260000e+00 -1.090000e+00   
25%    ...  3.000000e-02  3.000000e-02  2.000000e-02  2.000000e-02   
50%    ...  2.300000e-01  2.300000e-01  2.300000e-01  2.300000e-01   
75%    ...  7.505000e+00  7.500000e+00  7.300000e+00  7.100000e+00   
max    ...  2.951569e+06  3.070879e+06  3.169711e+06  3.246239e+06   

               2080          2085          2090          2095          2100  \
count  5.143600e+04  5.143600e+04  5.143600e+04  5.143600e+04  5.143600e+04   
mean   7.228327e+02  7.176899e+02  7.113072e+02  7.034274e+02  6.940296e+02   
std    2.455897e+04  2.496587e+04  2.530183e+04  2.556069e+04  2.574189e+04   
min   -9.200000e-01 -7.800000e-01 -6.500000e-01 -5.500000e-01 -4.500000e-01   
25%    1.000000e-02  1.000000e-02  1.000000e-02  1.000000e-02  1.000000e-02   
50%    2.300000e-01  2.300000e-01  2.300000e-01  2.300000e-01  2.200000e-01   
75%    6.722500e+00  6.080000e+00  5.462500e+00  4.680000e+00  4.032500e+00   
max    3.301586e+06  3.337871e+06  3.354746e+06  3.351887e+06  3.330484e+06   

       Unnamed: 69  
count          0.0  
mean           NaN  
std            NaN  
min            NaN  
25%            NaN  
50%            NaN  
75%            NaN  
max            NaN  

[8 rows x 66 columns]

7. Unique values for categorical columns of EdStatsData 

Unique values for Country Name: 242
['Arab World' 'East Asia & Pacific'
 'East Asia & Pacific (excluding high income)' 'Euro area'
 'Europe & Central Asia' 'Europe & Central Asia (excluding high income)'
 'European Union' 'Heavily indebted poor countries (HIPC)' 'High income'
 'Latin America & Caribbean'
 'Latin America & Caribbean (excluding high income)'
 'Least developed countries: UN classification' 'Low & middle income'
 'Low income' 'Lower middle income' 'Middle East & North Africa'
 'Middle East & North Africa (excluding high income)' 'Middle income'
 'North America' 'OECD members' 'South Asia' 'Sub-Saharan Africa'
 'Sub-Saharan Africa (excluding high income)' 'Upper middle income'
 'World' 'Afghanistan' 'Albania' 'Algeria' 'American Samoa' 'Andorra'
 'Angola' 'Antigua and Barbuda' 'Argentina' 'Armenia' 'Aruba' 'Australia'
 'Austria' 'Azerbaijan' 'Bahamas, The' 'Bahrain' 'Bangladesh' 'Barbados'
 'Belarus' 'Belgium' 'Beliz

> <u>Notes:</u>
>
> The EdStatsData dataset is a rich and voluminous set that covers 242 countries with 3,665 unique indicators, spanning 886,930 rows and 70 columns. The data extends over a period from 1970 to 2100, although some years between 2017 and 2100 are missing or grouped in five-year intervals.
>
> The first four columns, which are of object type, provide contextual information about the countries and indicators, while the subsequent columns, primarily of type float64, contain the indicator values for each year.
>
> A major challenge with this dataset is the significant amount of missing data. Specifically, 45 columns have more than 90% missing data, and 65 have more than 70% missing data. This will require special attention during the data cleaning phase. Additionally, a column, "Unnamed: 69", appears to be present by mistake and will need to be addressed.
>
> Despite these challenges, the diversity and depth of the data offer a valuable opportunity to extract relevant insights that can guide ACADEMY's strategic decisions regarding international expansion.


[Back to top](#top)

### 2 - <u>EdStatsCountry :</u><a id='stats-country'></a>

In [185]:
# Exploring EdStatsCountry
explore_df(COUNTRY, "EdStatsCountry")

Exploration of EdStatsCountry 

 

1. Preview the first and last line of EdStatsCountry 



Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   
2          AGO       Angola       Angola   People's Republic of Angola   
3          ALB      Albania      Albania           Republic of Albania   
4          AND      Andorra      Andorra       Principality of Andorra   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   
2           AO  Angolan kwanza   
3           AL    Albanian lek   
4           AD            Euro   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   
2  April 2013 database update: Based on IMF data,...   
3                                                NaN   
4                                                NaN   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   
2         Sub-Saharan Africa   Upper middle income        AO  ...   
3      Europe & Central Asia   Upper middle income        AL  ...   
4      Europe & Central Asia  High income: nonOECD        AD  ...   

            IMF data dissemination standard  \
0                                       NaN   
1  General Data Dissemination System (GDDS)   
2  General Data Dissemination System (GDDS)   
3  General Data Dissemination System (GDDS)   
4                                       NaN   

                            Latest population census  \
0                                               2010   
1                                               1979   
2                                               1970   
3                                               2011   
4  2011. Population figures compiled from adminis...   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   
2               Malaria Indicator Survey (MIS), 2011   
3       Demographic and Health Survey (DHS), 2008/09   
4                                                NaN   

   Source of most recent Income and expenditure data  \
0                                                NaN   
1            Integrated household survey (IHS), 2008   
2            Integrated household survey (IHS), 2008   
3  Living Standards Measurement Study Survey (LSM...   
4                                                NaN   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   
2                         NaN                       2015   
3                         Yes                       2012   
4                         Yes                        NaN   

  Latest industrial data Latest trade data Latest water withdrawal data  \
0                    NaN            2012.0                          NaN   
1                    NaN            2012.0                         2000   
2                    NaN               NaN                         2005   
3                 2010.0            2012.0                         2006   
4                    NaN            2006.0                          NaN   

  Unnamed: 31  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  

[5 rows x 32 columns]

Country Code    Short Name    Table Name                 Long Name  \
236          XKX        Kosovo        Kosovo        Republic of Kosovo   
237          YEM         Yemen   Yemen, Rep.         Republic of Yemen   
238          ZAF  South Africa  South Africa  Republic of South Africa   
239          ZMB        Zambia        Zambia        Republic of Zambia   
240          ZWE      Zimbabwe      Zimbabwe      Republic of Zimbabwe   

    2-alpha code       Currency Unit  \
236          NaN                Euro   
237           YE         Yemeni rial   
238           ZA  South African rand   
239           ZM  New Zambian kwacha   
240           ZW         U.S. dollar   

                                         Special Notes  \
236  Kosovo became a World Bank member on June 29, ...   
237  Based on official government statistics and In...   
238  Fiscal year end: March 31; reporting period fo...   
239  National accounts data have rebased to reflect...   
240  Fiscal year end: June 30; reporting period for...   

                         Region         Income Group WB-2 code  ...  \
236       Europe & Central Asia  Lower middle income        KV  ...   
237  Middle East & North Africa  Lower middle income        RY  ...   
238          Sub-Saharan Africa  Upper middle income        ZA  ...   
239          Sub-Saharan Africa  Lower middle income        ZM  ...   
240          Sub-Saharan Africa           Low income        ZW  ...   

                IMF data dissemination standard  Latest population census  \
236    General Data Dissemination System (GDDS)                      2011   
237    General Data Dissemination System (GDDS)                      2004   
238  Special Data Dissemination Standard (SDDS)                      2011   
239    General Data Dissemination System (GDDS)                      2010   
240    General Data Dissemination System (GDDS)                      2012   

                               Latest household survey  \
236                                                NaN   
237          Demographic and Health Survey (DHS), 2013   
238  Demographic and Health Survey (DHS), 2003; Wor...   
239          Demographic and Health Survey (DHS), 2013   
240       Demographic and Health Survey (DHS), 2010/11   

    Source of most recent Income and expenditure data  \
236           Integrated household survey (IHS), 2011   
237    Expenditure survey/budget survey (ES/BS), 2005   
238    Expenditure survey/budget survey (ES/BS), 2010   
239           Integrated household survey (IHS), 2010   
240        Integrated household survey (IHS), 2011/12   

    Vital registration complete            Latest agricultural census  \
236                         NaN                                   NaN   
237                         NaN                                   NaN   
238                         NaN                                  2007   
239                         NaN  2010. Population and Housing Census.   
240                         NaN                                   NaN   

    Latest industrial data Latest trade data Latest water withdrawal data  \
236                    NaN               NaN                          NaN   
237                 2006.0            2012.0                         2005   
238                 2010.0            2012.0                         2000   
239                    NaN            2011.0                         2002   
240                    NaN            2012.0                         2002   

    Unnamed: 31  
236         NaN  
237         NaN  
238         NaN  
239         NaN  
240         NaN  

[5 rows x 32 columns]

2. General information about EdStatsCountry 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 241 entries, 0 to 240
Data columns (total 32 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       241 non-null    object 
 1   Short Name                                         241 non-null    object 
 2   Table Name                                         241 non-null    object 
 3   Long Name                                          241 non-null    object 
 4   2-alpha code                                       238 non-null    object 
 5   Currency Unit                                      215 non-null    object 
 6   Special Notes                                      145 non-null    object 
 7   Region                                             214 non-null    object 
 8   Income Group                                

Missing Values  \
Unnamed: 31                                                   241   
National accounts reference year                              209   
Alternative conversion factor                                 194   
Other groups                                                  183   
Latest industrial data                                        134   
Vital registration complete                                   130   
External debt Reporting status                                117   
Latest household survey                                       100   
Latest agricultural census                                     99   
Lending category                                               97   
Special Notes                                                  96   
PPP survey year                                                96   
Source of most recent Income and expenditure data              81   
Government Accounting concept                                  80   
Latest water withdrawal data                                   62   
IMF data dissemination standard                                60   
Balance of Payments Manual in use                              60   
Latest trade data                                              56   
SNA price valuation                                            44   
System of trade                                                41   
National accounts base year                                    36   
Latest population census                                       28   
Income Group                                                   27   
Region                                                         27   
Currency Unit                                                  26   
System of National Accounts                                    26   
2-alpha code                                                    3   
WB-2 code                                                       1   

                                                   Missing Percentages (%)  
Unnamed: 31                                                         100.00  
National accounts reference year                                     86.72  
Alternative conversion factor                                        80.50  
Other groups                                                         75.93  
Latest industrial data                                               55.60  
Vital registration complete                                          53.94  
External debt Reporting status                                       48.55  
Latest household survey                                              41.49  
Latest agricultural census                                           41.08  
Lending category                                                     40.25  
Special Notes                                                        39.83  
PPP survey year                                                      39.83  
Source of most recent Income and expenditure data                    33.61  
Government Accounting concept                                        33.20  
Latest water withdrawal data                                         25.73  
IMF data dissemination standard                                      24.90  
Balance of Payments Manual in use                                    24.90  
Latest trade data                                                    23.24  
SNA price valuation                                                  18.26  
System of trade                                                      17.01  
National accounts base year                                          14.94  
Latest population census                                             11.62  
Income Group                                                         11.20  
Region                                                               11.20  
Currency Unit                                                        10.79  
System of National Accounts                    

5. Duplicate rows in EdStatsCountry 

There are no duplicate rows.
6. Descriptive statistics for EdStatsCountry 



National accounts reference year  Latest industrial data  \
count                          32.00000              107.000000   
mean                         2001.53125             2008.102804   
std                             5.24856                2.616834   
min                          1987.00000             2000.000000   
25%                          1996.75000             2007.500000   
50%                          2002.00000             2009.000000   
75%                          2005.00000             2010.000000   
max                          2012.00000             2010.000000   

       Latest trade data  Unnamed: 31  
count         185.000000          0.0  
mean         2010.994595          NaN  
std             2.569675          NaN  
min          1995.000000          NaN  
25%          2011.000000          NaN  
50%          2012.000000          NaN  
75%          2012.000000          NaN  
max          2012.000000          NaN

7. Unique values for categorical columns of EdStatsCountry 

Unique values for Country Code: 241
['ABW' 'AFG' 'AGO' 'ALB' 'AND' 'ARB' 'ARE' 'ARG' 'ARM' 'ASM' 'ATG' 'AUS'
 'AUT' 'AZE' 'BDI' 'BEL' 'BEN' 'BFA' 'BGD' 'BGR' 'BHR' 'BHS' 'BIH' 'BLR'
 'BLZ' 'BMU' 'BOL' 'BRA' 'BRB' 'BRN' 'BTN' 'BWA' 'CAF' 'CAN' 'CHE' 'CHI'
 'CHL' 'CHN' 'CIV' 'CMR' 'COD' 'COG' 'COL' 'COM' 'CPV' 'CRI' 'CUB' 'CUW'
 'CYM' 'CYP' 'CZE' 'DEU' 'DJI' 'DMA' 'DNK' 'DOM' 'DZA' 'EAP' 'EAS' 'ECA'
 'ECS' 'ECU' 'EGY' 'EMU' 'ERI' 'ESP' 'EST' 'ETH' 'EUU' 'FIN' 'FJI' 'FRA'
 'FRO' 'FSM' 'GAB' 'GBR' 'GEO' 'GHA' 'GIB' 'GIN' 'GMB' 'GNB' 'GNQ' 'GRC'
 'GRD' 'GRL' 'GTM' 'GUM' 'GUY' 'HIC' 'HKG' 'HND' 'HPC' 'HRV' 'HTI' 'HUN'
 'IDN' 'IMN' 'IND' 'IRL' 'IRN' 'IRQ' 'ISL' 'ISR' 'ITA' 'JAM' 'JOR' 'JPN'
 'KAZ' 'KEN' 'KGZ' 'KHM' 'KIR' 'KNA' 'KOR' 'KWT' 'LAC' 'LAO' 'LBN' 'LBR'
 'LBY' 'LCA' 'LCN' 'LDC' 'LIC' 'LIE' 'LKA' 'LMC' 'LMY' 'LSO' 'LTU' 'LUX'
 'LVA' 'MAC' 'MAF' 'MAR' 'MCO' 'MDA' 'MDG' 'MDV' 'MEA' 'MEX' 'MHL' 'MIC'
 'MKD' 'MLI' 'MLT' 'MMR' 'M

> <u>Notes:</u>
>
> The EdStatsCountry dataset provides metadata on 241 countries, covering aspects such as country codes, names, regions, and income groups. While the dataset contains 32 columns, some of them, like "Unnamed: 31", appear to be redundant or entirely empty. Special attention will need to be given to these columns during the data cleaning phase. The presence of 7 unique regions offers a perspective for grouping and analyzing countries based on geographical criteria.


[Back to top](#top)

### 3 - <u>EdStatsCountry-Series :</u><a id='stats-country-series'></a>

In [186]:
# Exploring EdStatsCountry-Series
explore_df(COUNTRY_SERIES, "EdStatsCountry-Series")

Exploration of EdStatsCountry-Series 

 

1. Preview the first and last line of EdStatsCountry-Series 



CountryCode         SeriesCode  \
0         ABW        SP.POP.TOTL   
1         ABW        SP.POP.GROW   
2         AFG        SP.POP.GROW   
3         AFG  NY.GDP.PCAP.PP.CD   
4         AFG        SP.POP.TOTL   

                                         DESCRIPTION  Unnamed: 3  
0  Data sources : United Nations World Population...         NaN  
1  Data sources: United Nations World Population ...         NaN  
2  Data sources: United Nations World Population ...         NaN  
3                 Estimates are based on regression.         NaN  
4  Data sources : United Nations World Population...         NaN

CountryCode   SeriesCode  \
608         ZAF  SP.POP.GROW   
609         ZMB  SP.POP.GROW   
610         ZMB  SP.POP.TOTL   
611         ZWE  SP.POP.TOTL   
612         ZWE  SP.POP.GROW   

                                           DESCRIPTION  Unnamed: 3  
608  Data sources : Statistics South Africa, United...         NaN  
609  Data sources: United Nations World Population ...         NaN  
610  Data sources : United Nations World Population...         NaN  
611  Data sources : United Nations World Population...         NaN  
612  Data sources: United Nations World Population ...         NaN

2. General information about EdStatsCountry-Series 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 613 entries, 0 to 612
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   CountryCode  613 non-null    object 
 1   SeriesCode   613 non-null    object 
 2   DESCRIPTION  613 non-null    object 
 3   Unnamed: 3   0 non-null      float64
dtypes: float64(1), object(3)
memory usage: 19.3+ KB
3. Data types in EdStatsCountry-Series 

object     3
float64    1
dtype: int64
4. Missing values in EdStatsCountry-Series 

Displaying all columns with missing data: 



Missing Values  Missing Percentages (%)
Unnamed: 3             613                    100.0

5. Duplicate rows in EdStatsCountry-Series 

There are no duplicate rows.
6. Descriptive statistics for EdStatsCountry-Series 



Unnamed: 3
count         0.0
mean          NaN
std           NaN
min           NaN
25%           NaN
50%           NaN
75%           NaN
max           NaN

7. Unique values for categorical columns of EdStatsCountry-Series 

Unique values for CountryCode: 211
['ABW' 'AFG' 'AGO' 'ALB' 'AND' 'ARE' 'ARG' 'ARM' 'ASM' 'ATG' 'AUS' 'AUT'
 'AZE' 'BDI' 'BEL' 'BEN' 'BFA' 'BGD' 'BGR' 'BHR' 'BHS' 'BIH' 'BLR' 'BLZ'
 'BMU' 'BOL' 'BRA' 'BRB' 'BRN' 'BTN' 'BWA' 'CAF' 'CAN' 'CHE' 'CHI' 'CHL'
 'CHN' 'CIV' 'CMR' 'COD' 'COG' 'COL' 'COM' 'CPV' 'CRI' 'CUB' 'CYM' 'CYP'
 'CZE' 'DEU' 'DJI' 'DMA' 'DNK' 'DOM' 'DZA' 'ECU' 'EGY' 'ERI' 'ESP' 'EST'
 'ETH' 'FIN' 'FJI' 'FRA' 'FRO' 'FSM' 'GAB' 'GBR' 'GEO' 'GHA' 'GIB' 'GIN'
 'GMB' 'GNB' 'GNQ' 'GRC' 'GRD' 'GRL' 'GTM' 'GUM' 'GUY' 'HKG' 'HND' 'HRV'
 'HTI' 'HUN' 'IDN' 'IMN' 'IND' 'IRL' 'IRN' 'IRQ' 'ISL' 'ISR' 'ITA' 'JAM'
 'JOR' 'JPN' 'KAZ' 'KEN' 'KGZ' 'KHM' 'KIR' 'KNA' 'KOR' 'KWT' 'LAO' 'LBN'
 'LBR' 'LBY' 'LCA' 'LIE' 'LKA' 'LSO' 'LTU' 'LUX' 'LVA' 'MAC' 'MAR' 'MCO'
 'MDA' 'MDG' 'MDV' 'MEX' 'MHL' 'MKD' 'MLI' 'MLT' 'MMR' 'MNE' 'MNG' 'MNP'
 'MOZ' 'MRT' 'MUS' 'MWI' 'MYS' 'NAM' 'NCL' 'NER' 'NGA' 'NIC' 'NLD' 'NOR'
 'NPL' 'NZL' 'OMN' 'P

> <u>Notes:</u>
>
> The EdStatsCountry-Series dataset contains metadata related to 211 countries and 21 unique series codes. Although the dataset is relatively smaller with 613 rows and 4 columns, it provides valuable information about the source of indicators for each country. The "Unnamed: 3" column appears to be an accidental addition and will need to be addressed during the data cleaning phase.



[Back to top](#top)

### 4 - <u>EdStatsSeries :</u><a id='stats-series'></a>

In [187]:
# Exploring EdStatsSeries
explore_df(SERIES, "EdStatsSeries")

Exploration of EdStatsSeries 

 

1. Preview the first and last line of EdStatsSeries 



Series Code       Topic  \
0  BAR.NOED.1519.FE.ZS  Attainment   
1     BAR.NOED.1519.ZS  Attainment   
2  BAR.NOED.15UP.FE.ZS  Attainment   
3     BAR.NOED.15UP.ZS  Attainment   
4  BAR.NOED.2024.FE.ZS  Attainment   

                                      Indicator Name  \
0  Barro-Lee: Percentage of female population age...   
1  Barro-Lee: Percentage of population age 15-19 ...   
2  Barro-Lee: Percentage of female population age...   
3  Barro-Lee: Percentage of population age 15+ wi...   
4  Barro-Lee: Percentage of female population age...   

                                    Short definition  \
0  Percentage of female population age 15-19 with...   
1  Percentage of population age 15-19 with no edu...   
2  Percentage of female population age 15+ with n...   
3  Percentage of population age 15+ with no educa...   
4  Percentage of female population age 20-24 with...   

                                     Long definition  Unit of measure  \
0  Percentage of female population age 15-19 with...              NaN   
1  Percentage of population age 15-19 with no edu...              NaN   
2  Percentage of female population age 15+ with n...              NaN   
3  Percentage of population age 15+ with no educa...              NaN   
4  Percentage of female population age 20-24 with...              NaN   

  Periodicity Base Period Other notes Aggregation method  ...  \
0         NaN         NaN         NaN                NaN  ...   
1         NaN         NaN         NaN                NaN  ...   
2         NaN         NaN         NaN                NaN  ...   
3         NaN         NaN         NaN                NaN  ...   
4         NaN         NaN         NaN                NaN  ...   

  Notes from original source  General comments  \
0                        NaN               NaN   
1                        NaN               NaN   
2                        NaN               NaN   
3                        NaN               NaN   
4                        NaN               NaN   

                                              Source  \
0  Robert J. Barro and Jong-Wha Lee: http://www.b...   
1  Robert J. Barro and Jong-Wha Lee: http://www.b...   
2  Robert J. Barro and Jong-Wha Lee: http://www.b...   
3  Robert J. Barro and Jong-Wha Lee: http://www.b...   
4  Robert J. Barro and Jong-Wha Lee: http://www.b...   

  Statistical concept and methodology Development relevance  \
0                                 NaN                   NaN   
1                                 NaN                   NaN   
2                                 NaN                   NaN   
3                                 NaN                   NaN   
4                                 NaN                   NaN   

  Related source links Other web links  Related indicators  License Type  \
0                  NaN             NaN                 NaN           NaN   
1                  NaN             NaN                 NaN           NaN   
2                  NaN             NaN                 NaN           NaN   
3                  NaN             NaN                 NaN           NaN   
4                  NaN             NaN                 NaN           NaN   

   Unnamed: 20  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN  

[5 rows x 21 columns]

Series Code         Topic  \
3660    UIS.XUNIT.USCONST.3.FSGOV  Expenditures   
3661    UIS.XUNIT.USCONST.4.FSGOV  Expenditures   
3662   UIS.XUNIT.USCONST.56.FSGOV  Expenditures   
3663  XGDP.23.FSGOV.FDINSTADM.FFD  Expenditures   
3664  XGDP.56.FSGOV.FDINSTADM.FFD  Expenditures   

                                         Indicator Name  \
3660  Government expenditure per upper secondary stu...   
3661  Government expenditure per post-secondary non-...   
3662  Government expenditure per tertiary student (c...   
3663  Government expenditure in secondary institutio...   
3664  Government expenditure in tertiary institution...   

                                       Short definition  \
3660                                                NaN   
3661                                                NaN   
3662                                                NaN   
3663  Total general (local, regional and central) go...   
3664  Total general (local, regional and central) go...   

                                        Long definition  Unit of measure  \
3660  Average total (current, capital and transfers)...              NaN   
3661  Average total (current, capital and transfers)...              NaN   
3662  Average total (current, capital and transfers)...              NaN   
3663  Total general (local, regional and central) go...              NaN   
3664  Total general (local, regional and central) go...              NaN   

     Periodicity Base Period Other notes Aggregation method  ...  \
3660         NaN         NaN         NaN                NaN  ...   
3661         NaN         NaN         NaN                NaN  ...   
3662         NaN         NaN         NaN                NaN  ...   
3663         NaN         NaN   Secondary                NaN  ...   
3664         NaN         NaN    Tertiary                NaN  ...   

     Notes from original source  General comments  \
3660                        NaN               NaN   
3661                        NaN               NaN   
3662                        NaN               NaN   
3663                        NaN               NaN   
3664                        NaN               NaN   

                               Source Statistical concept and methodology  \
3660  UNESCO Institute for Statistics                                 NaN   
3661  UNESCO Institute for Statistics                                 NaN   
3662  UNESCO Institute for Statistics                                 NaN   
3663  UNESCO Institute for Statistics                                 NaN   
3664  UNESCO Institute for Statistics                                 NaN   

     Development relevance Related source links Other web links  \
3660                   NaN                  NaN             NaN   
3661                   NaN                  NaN             NaN   
3662                   NaN                  NaN             NaN   
3663                   NaN                  NaN             NaN   
3664                   NaN                  NaN             NaN   

      Related indicators  License Type  Unnamed: 20  
3660                 NaN           NaN          NaN  
3661                 NaN           NaN          NaN  
3662                 NaN           NaN          NaN  
3663                 NaN           NaN          NaN  
3664                 NaN           NaN          NaN  

[5 rows x 21 columns]

2. General information about EdStatsSeries 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3665 entries, 0 to 3664
Data columns (total 21 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Series Code                          3665 non-null   object 
 1   Topic                                3665 non-null   object 
 2   Indicator Name                       3665 non-null   object 
 3   Short definition                     2156 non-null   object 
 4   Long definition                      3665 non-null   object 
 5   Unit of measure                      0 non-null      float64
 6   Periodicity                          99 non-null     object 
 7   Base Period                          314 non-null    object 
 8   Other notes                          552 non-null    object 
 9   Aggregation method                   47 non-null     object 
 10  Limitations and exceptions           14 non-null   

Missing Values  Missing Percentages (%)
Unit of measure                                3665                   100.00
Notes from original source                     3665                   100.00
Other web links                                3665                   100.00
Related indicators                             3665                   100.00
License Type                                   3665                   100.00
Unnamed: 20                                    3665                   100.00
Development relevance                          3662                    99.92
Limitations and exceptions                     3651                    99.62
General comments                               3651                    99.62
Statistical concept and methodology            3642                    99.37
Aggregation method                             3618                    98.72
Periodicity                                    3566                    97.30
Related source links                           3450                    94.13
Base Period                                    3351                    91.43
Other notes                                    3113                    84.94
Short definition                               1509                    41.17

5. Duplicate rows in EdStatsSeries 

There are no duplicate rows.
6. Descriptive statistics for EdStatsSeries 



Unit of measure  Notes from original source  Other web links  \
count              0.0                         0.0              0.0   
mean               NaN                         NaN              NaN   
std                NaN                         NaN              NaN   
min                NaN                         NaN              NaN   
25%                NaN                         NaN              NaN   
50%                NaN                         NaN              NaN   
75%                NaN                         NaN              NaN   
max                NaN                         NaN              NaN   

       Related indicators  License Type  Unnamed: 20  
count                 0.0           0.0          0.0  
mean                  NaN           NaN          NaN  
std                   NaN           NaN          NaN  
min                   NaN           NaN          NaN  
25%                   NaN           NaN          NaN  
50%                   NaN           NaN          NaN  
75%                   NaN           NaN          NaN  
max                   NaN           NaN          NaN

7. Unique values for categorical columns of EdStatsSeries 

Unique values for Series Code: 3665
['BAR.NOED.1519.FE.ZS' 'BAR.NOED.1519.ZS' 'BAR.NOED.15UP.FE.ZS' ...
 'UIS.XUNIT.USCONST.56.FSGOV' 'XGDP.23.FSGOV.FDINSTADM.FFD'
 'XGDP.56.FSGOV.FDINSTADM.FFD']


Unique values for Topic: 37
['Attainment' 'Education Equality' 'Infrastructure: Communications'
 'Learning Outcomes'
 'Economic Policy & Debt: National accounts: US$ at current prices: Aggregate indicators'
 'Economic Policy & Debt: National accounts: US$ at constant 2010 prices: Aggregate indicators'
 'Economic Policy & Debt: Purchasing power parity'
 'Economic Policy & Debt: National accounts: Atlas GNI & GNI per capita'
 'Teachers' 'Education Management Information Systems (SABER)'
 'Early Child Development (SABER)' 'Engaging the Private Sector (SABER)'
 'School Health and School Feeding (SABER)'
 'School Autonomy and Accountability (SABER)' 'School Finance (SABER)'
 'Student Assessment (SABER)' 'Teachers (SABER)'
 'Tertiary Educ

> <u>Notes:</u>
>
>EdStatsSeries is a dataset that provides details on 3,665 unique indicators, covering various topics and themes. With 3,665 rows and 21 columns, this dataset is essential for understanding the meaning and context of each indicator. However, several columns, including "Unnamed: 20", appear to contain a large amount of missing data or be redundant. These columns will need to be closely examined during the data cleaning phase.

[Back to top](#top)

### 5 - <u>EdStatsFootNote :</u><a id='stats-foot-note'></a>

In [188]:
# Exploring EdStatsFootNote
explore_df(FOOT_NOTE, "EdStatsFootNote")

Exploration of EdStatsFootNote 

 

1. Preview the first and last line of EdStatsFootNote 



CountryCode      SeriesCode    Year          DESCRIPTION  Unnamed: 4
0         ABW  SE.PRE.ENRL.FE  YR2001  Country estimation.         NaN
1         ABW  SE.TER.TCHR.FE  YR2005  Country estimation.         NaN
2         ABW  SE.PRE.TCHR.FE  YR2000  Country estimation.         NaN
3         ABW  SE.SEC.ENRL.GC  YR2004  Country estimation.         NaN
4         ABW     SE.PRE.TCHR  YR2006  Country estimation.         NaN

CountryCode   SeriesCode    Year  \
643633         ZWE  SH.DYN.MORT  YR2007   
643634         ZWE  SH.DYN.MORT  YR2014   
643635         ZWE  SH.DYN.MORT  YR2015   
643636         ZWE  SH.DYN.MORT  YR2017   
643637         ZWE  SP.POP.GROW  YR2017   

                                   DESCRIPTION  Unnamed: 4  
643633       Uncertainty bound is 91.6 - 109.3         NaN  
643634          Uncertainty bound is 54.3 - 76         NaN  
643635        Uncertainty bound is 48.3 - 73.3         NaN  
643636  5-year average value between 0s and 5s         NaN  
643637  5-year average value between 0s and 5s         NaN

2. General information about EdStatsFootNote 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 643638 entries, 0 to 643637
Data columns (total 5 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   CountryCode  643638 non-null  object 
 1   SeriesCode   643638 non-null  object 
 2   Year         643638 non-null  object 
 3   DESCRIPTION  643638 non-null  object 
 4   Unnamed: 4   0 non-null       float64
dtypes: float64(1), object(4)
memory usage: 24.6+ MB
3. Data types in EdStatsFootNote 

object     4
float64    1
dtype: int64
4. Missing values in EdStatsFootNote 

Displaying all columns with missing data: 



Missing Values  Missing Percentages (%)
Unnamed: 4          643638                    100.0

5. Duplicate rows in EdStatsFootNote 

There are no duplicate rows.
6. Descriptive statistics for EdStatsFootNote 



Unnamed: 4
count         0.0
mean          NaN
std           NaN
min           NaN
25%           NaN
50%           NaN
75%           NaN
max           NaN

7. Unique values for categorical columns of EdStatsFootNote 

Unique values for CountryCode: 239
['ABW' 'AFG' 'AGO' 'ALB' 'AND' 'ARB' 'ARE' 'ARG' 'ARM' 'ASM' 'ATG' 'AUS'
 'AUT' 'AZE' 'BDI' 'BEL' 'BEN' 'BFA' 'BGD' 'BGR' 'BHR' 'BHS' 'BIH' 'BLR'
 'BLZ' 'BMU' 'BOL' 'BRA' 'BRB' 'BRN' 'BTN' 'BWA' 'CAF' 'CAN' 'CHE' 'CHI'
 'CHL' 'CHN' 'CIV' 'CMR' 'COD' 'COG' 'COL' 'COM' 'CPV' 'CRI' 'CUB' 'CUW'
 'CYM' 'CYP' 'CZE' 'DEU' 'DJI' 'DMA' 'DNK' 'DOM' 'DZA' 'EAP' 'EAS' 'ECA'
 'ECS' 'ECU' 'EGY' 'EMU' 'ERI' 'ESP' 'EST' 'ETH' 'EUU' 'FIN' 'FJI' 'FRA'
 'FRO' 'FSM' 'GAB' 'GBR' 'GEO' 'GHA' 'GIB' 'GIN' 'GMB' 'GNB' 'GNQ' 'GRC'
 'GRD' 'GTM' 'GUM' 'GUY' 'HIC' 'HKG' 'HND' 'HPC' 'HRV' 'HTI' 'HUN' 'IDN'
 'IMN' 'IND' 'IRL' 'IRN' 'IRQ' 'ISL' 'ISR' 'ITA' 'JAM' 'JOR' 'JPN' 'KAZ'
 'KEN' 'KGZ' 'KHM' 'KIR' 'KNA' 'KOR' 'KWT' 'LAC' 'LAO' 'LBN' 'LBR' 'LBY'
 'LCA' 'LCN' 'LDC' 'LIC' 'LIE' 'LKA' 'LMC' 'LMY' 'LSO' 'LTU' 'LUX' 'LVA'
 'MAC' 'MAR' 'MCO' 'MDA' 'MDG' 'MDV' 'MEA' 'MEX' 'MHL' 'MIC' 'MKD' 'MLI'
 'MLT' 'MMR' 'MNA' 'MNE' 'M

> <u>Notes:</u>
>
> The EdStatsFootNote dataset provides additional notes for the indicators, offering context and further details for 643,638 unique entries. While the dataset contains 5 columns, the "Unnamed: 4" column appears to be redundant and entirely empty. This column, along with other potential anomalies, will need to be addressed during the data cleaning phase.

[Back to top](#top)

### <u>Analysis note:</u><a id='analysis-note'></a>

#### 1. sData
- **Country Name**: Name of the countries.
- **Country Code**: Unique code of the country.
- **Indicator Name**: Name of the indicator.
- **Indicator Code**: Unique code of the indicator.
- **Years**: Years corresponding to each data point.

#### 2. sCountry
- **Country Code**: Unique code of the country.
- **Table Name**: Country name as per ISO standards.
- **Region**: Geographical region of the country.
- **Income Group**: Country's income category.

#### 3. sCountry Serie
- **Country Code**: Unique code of the country.
- **Series Code**: Unique code of the indicator.
- **Description**: Data source of the indicator.

#### 4. sSerie
- **Series Code**: Unique code of the indicator.
- **Topic**: Subject or theme of the indicator.
- **Indicator Name**: Name of the indicator.

#### 5. sFoot
Contains similar information to sCountry, but with the added year of the indicator and a detailed description of the data source.


#### General Observations
- No duplicate rows were found across all files.
- Several columns have a significant number of missing values, especially those with a 100% missing rate. I think it's recommended to remove these columns during the data cleaning phase.


#### Conclusion
Based on this information, the most relevant columns for our Ed-Tech study are: Country Name, Indicator Code, Country Code, Region, Income Group, Series Code, and Years.

[Back to top](#top)

## Data Cleaning <a id='data-cleaning'></a>
> Address any issues in the dataset to ensure its quality and reliability. This involves handling missing values, duplicates, and any inconsistencies.

### Relevant Indicators <a id='relevant indicators'></a>

To access the courses offered, potential clients must have a computer and an internet connection. Our target audience is students in upper secondary school and tertiary education, specifically those aged 15 to 24.

Here's a breakdown of the indicators:
- **UIS.E.3**: This represents the total number of students enrolled in both public and private upper secondary education institutions, irrespective of their age.

- **IT.CMP.PCMP.P2**: This indicates the number of personal computers per 100 people. Personal computers are standalone devices intended for individual use.

- **IT.NET.USER.P2**: This represents the number of internet users per 100 people. Internet users are defined as individuals who have accessed the internet (from any location) in the past 3 months. The internet can be accessed via various devices such as computers, mobile phones, PDAs, gaming consoles, digital TVs, etc.

- **SE.TER.ENRL**: This is the total number of students enrolled in both public and private tertiary education institutions.

- **SE.XPD.TOTL.GD.ZS**: This indicates the total expenditure (current, capital, and transfers) by local, regional, and central governments on education, expressed as a percentage of the GDP.

- **SP.POP.1524.TO.UN**: This represents the total population aged between 15 and 24.

In conclusion, we have identified 6 key indicators that will be instrumental for our analysis.


In [189]:
# List of relevant indicators for the analysis
relevant_indicators = [
    'SE.TER.ENRL',        # Total students enrolled in tertiary education
    'UIS.E.3',            # Total students enrolled in upper secondary education
    'IT.NET.USER.P2',     # Internet users per 100 people
    'IT.CMP.PCMP.P2',     # Personal computers per 100 people
    'SE.XPD.TOTL.GD.ZS',  # Government expenditure on education as % of GDP
    'SP.POP.1524.TO.UN'   # Total population aged between 15 and 24
]

### Filter & Eliminate unnecessary columns <a id='filter-&-eliminate-unnecessary-columns'></a>

#### 1 - <u>EdStatsData</u> <a id='cleaning-stats-data'></a>

In [190]:
# Filter the data DataFrame based on the relevant indicators
data = data[data['Indicator Code'].isin(relevant_indicators)]

In [191]:
# Display data
data

Country Name Country Code  \
1204     Arab World          ARB   
1214     Arab World          ARB   
1260     Arab World          ARB   
1375     Arab World          ARB   
2084     Arab World          ARB   
...             ...          ...   
884479     Zimbabwe          ZWE   
884525     Zimbabwe          ZWE   
884640     Zimbabwe          ZWE   
885349     Zimbabwe          ZWE   
885747     Zimbabwe          ZWE   

                                           Indicator Name     Indicator Code  \
1204    Enrolment in tertiary education, all programme...        SE.TER.ENRL   
1214    Enrolment in upper secondary education, both s...            UIS.E.3   
1260    Government expenditure on education as % of GD...  SE.XPD.TOTL.GD.ZS   
1375                      Internet users (per 100 people)     IT.NET.USER.P2   
2084                  Personal computers (per 100 people)     IT.CMP.PCMP.P2   
...                                                   ...                ...   
884479  Enrolment in upper secondary education, both s...            UIS.E.3   
884525  Government expenditure on education as % of GD...  SE.XPD.TOTL.GD.ZS   
884640                    Internet users (per 100 people)     IT.NET.USER.P2   
885349                Personal computers (per 100 people)     IT.CMP.PCMP.P2   
885747                      Population, ages 15-24, total  SP.POP.1524.TO.UN   

              1970       1971      1972         1973        1974        1975  \
1204    706416.125  733981.25  794759.0  871347.3125  957383.125  1066646.75   
1214           NaN        NaN       NaN          NaN         NaN         NaN   
1260           NaN        NaN       NaN          NaN         NaN         NaN   
1375           NaN        NaN       NaN          NaN         NaN         NaN   
2084           NaN        NaN       NaN          NaN         NaN         NaN   
...            ...        ...       ...          ...         ...         ...   
884479         NaN        NaN       NaN          NaN         NaN         NaN   
884525         NaN        NaN       NaN          NaN         NaN         NaN   
884640         NaN        NaN       NaN          NaN         NaN         NaN   
885349         NaN        NaN       NaN          NaN         NaN         NaN   
885747         NaN        NaN       NaN          NaN         NaN         NaN   

        ...  2060  2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
1204    ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1214    ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1260    ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1375    ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
2084    ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
...     ...   ...   ...   ...   ...   ...   ...   ...   ...   ...          ...  
884479  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
884525  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
884640  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
885349  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
885747  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[1452 rows x 70 columns]

In [192]:
# Display missing data == 70%
missing_values_table(data, 70)

Displaying columns with missing data >= 70%: 

Total columns that have missing data >= 70%: 40


Missing Values  Missing Percentages (%)
Unnamed: 69            1452                   100.00
2060                   1452                   100.00
2025                   1452                   100.00
2030                   1452                   100.00
2035                   1452                   100.00
2040                   1452                   100.00
2045                   1452                   100.00
2050                   1452                   100.00
2055                   1452                   100.00
2065                   1452                   100.00
2017                   1452                   100.00
2070                   1452                   100.00
2075                   1452                   100.00
2080                   1452                   100.00
2085                   1452                   100.00
2090                   1452                   100.00
2095                   1452                   100.00
2100                   1452                   100.00
2020                   1452                   100.00
1970                   1354                    93.25
1974                   1273                    87.67
1972                   1271                    87.53
1973                   1267                    87.26
1976                   1261                    86.85
1977                   1260                    86.78
1978                   1259                    86.71
1971                   1258                    86.64
1975                   1254                    86.36
1979                   1244                    85.67
1983                   1242                    85.54
1987                   1240                    85.40
1985                   1240                    85.40
1980                   1235                    85.06
1984                   1235                    85.06
1982                   1233                    84.92
1981                   1222                    84.16
1986                   1220                    84.02
2016                   1218                    83.88
1988                   1193                    82.16
1989                   1183                    81.47

In [193]:
# Remove columns that contain +70% missing values
drop_columns(data, 70)

Displaying all columns with missing data: 

40 columns dropped because it they had 70% missing values: Unnamed: 69, 2060, 2025, 2030, 2035, 2040, 2045, 2050, 2055, 2065, 2017, 2070, 2075, 2080, 2085, 2090, 2095, 2100, 2020, 1970, 1974, 1972, 1973, 1976, 1977, 1978, 1971, 1975, 1979, 1983, 1987, 1985, 1984, 1980, 1982, 1981, 1986, 2016, 1988, 1989


In [194]:
# Displaying all columns with missing data filtered by percentage >= 70%
missing_values_table(data)

Displaying all columns with missing data: 



Missing Values  Missing Percentages (%)
1991             943                    64.94
1992             921                    63.43
1993             894                    61.57
2015             880                    60.61
1994             835                    57.51
1990             762                    52.48
1995             750                    51.65
1997             742                    51.10
1996             711                    48.97
2014             667                    45.94
1998             649                    44.70
2013             606                    41.74
2012             588                    40.50
2009             580                    39.94
2010             577                    39.74
2011             575                    39.60
2008             557                    38.36
2007             546                    37.60
2006             485                    33.40
2000             411                    28.31
1999             409                    28.17
2003             396                    27.27
2005             394                    27.13
2001             393                    27.07
2002             371                    25.55
2004             370                    25.48

In [195]:
print(title("After Filtering Out Irrelevant Data:"))
print(f"- Number of rows: {data.shape[0]}")
print(f"- Number of columns: {data.shape[1]}")

After Filtering Out Irrelevant Data: 

- Number of rows: 1452
- Number of columns: 30


#### 2 - <u>EdStatsCountry</u> <a id='cleaning-stats-country'></a>

In [196]:
# Display columns names
COUNTRY.columns.tolist()

['Country Code',
 'Short Name',
 'Table Name',
 'Long Name',
 '2-alpha code',
 'Currency Unit',
 'Special Notes',
 'Region',
 'Income Group',
 'WB-2 code',
 'National accounts base year',
 'National accounts reference year',
 'SNA price valuation',
 'Lending category',
 'Other groups',
 'System of National Accounts',
 'Alternative conversion factor',
 'PPP survey year',
 'Balance of Payments Manual in use',
 'External debt Reporting status',
 'System of trade',
 'Government Accounting concept',
 'IMF data dissemination standard',
 'Latest population census',
 'Latest household survey',
 'Source of most recent Income and expenditure data',
 'Vital registration complete',
 'Latest agricultural census',
 'Latest industrial data',
 'Latest trade data',
 'Latest water withdrawal data',
 'Unnamed: 31']

In [197]:
# Keep only relevant columns
country = country[['Country Code', 'Short Name', 'Table Name', 'Long Name']]

In [198]:
# Display table of 'country'
country

Country Code    Short Name    Table Name                     Long Name
0            ABW         Aruba         Aruba                         Aruba
1            AFG   Afghanistan   Afghanistan  Islamic State of Afghanistan
2            AGO        Angola        Angola   People's Republic of Angola
3            ALB       Albania       Albania           Republic of Albania
4            AND       Andorra       Andorra       Principality of Andorra
..           ...           ...           ...                           ...
236          XKX        Kosovo        Kosovo            Republic of Kosovo
237          YEM         Yemen   Yemen, Rep.             Republic of Yemen
238          ZAF  South Africa  South Africa      Republic of South Africa
239          ZMB        Zambia        Zambia            Republic of Zambia
240          ZWE      Zimbabwe      Zimbabwe          Republic of Zimbabwe

[241 rows x 4 columns]

In [199]:
missing_values_table(country)

Displaying all columns with missing data: 

There are no columns with the specified percentage of missing data.


In [200]:
print(title("After Filtering Out Irrelevant Data:"))
print(f"- Number of rows: {country.shape[0]}")
print(f"- Number of columns: {country.shape[1]}")

After Filtering Out Irrelevant Data: 

- Number of rows: 241
- Number of columns: 4


In [201]:
country_series

CountryCode         SeriesCode  \
0           ABW        SP.POP.TOTL   
1           ABW        SP.POP.GROW   
2           AFG        SP.POP.GROW   
3           AFG  NY.GDP.PCAP.PP.CD   
4           AFG        SP.POP.TOTL   
..          ...                ...   
608         ZAF        SP.POP.GROW   
609         ZMB        SP.POP.GROW   
610         ZMB        SP.POP.TOTL   
611         ZWE        SP.POP.TOTL   
612         ZWE        SP.POP.GROW   

                                           DESCRIPTION  Unnamed: 3  
0    Data sources : United Nations World Population...         NaN  
1    Data sources: United Nations World Population ...         NaN  
2    Data sources: United Nations World Population ...         NaN  
3                   Estimates are based on regression.         NaN  
4    Data sources : United Nations World Population...         NaN  
..                                                 ...         ...  
608  Data sources : Statistics South Africa, United...         NaN  
609  Data sources: United Nations World Population ...         NaN  
610  Data sources : United Nations World Population...         NaN  
611  Data sources : United Nations World Population...         NaN  
612  Data sources: United Nations World Population ...         NaN  

[613 rows x 4 columns]

#### 3 - <u>EdStatsCountrySeries</u> <a id='cleaning-stats-country-series'></a>

In [202]:
# Display columns names
country_series.columns.tolist()

['CountryCode', 'SeriesCode', 'DESCRIPTION', 'Unnamed: 3']

In [203]:
missing_values_table(country_series)

Displaying all columns with missing data: 



Missing Values  Missing Percentages (%)
Unnamed: 3             613                    100.0

In [204]:
# Drop columns that contain +100% missing values
drop_columns(country_series, 100)

Displaying all columns with missing data: 

1 column dropped because it had 100% missing values: Unnamed: 3


In [205]:
# Displaying the table with missing data filtered by percentage >= 100%
country_series

CountryCode         SeriesCode  \
0           ABW        SP.POP.TOTL   
1           ABW        SP.POP.GROW   
2           AFG        SP.POP.GROW   
3           AFG  NY.GDP.PCAP.PP.CD   
4           AFG        SP.POP.TOTL   
..          ...                ...   
608         ZAF        SP.POP.GROW   
609         ZMB        SP.POP.GROW   
610         ZMB        SP.POP.TOTL   
611         ZWE        SP.POP.TOTL   
612         ZWE        SP.POP.GROW   

                                           DESCRIPTION  
0    Data sources : United Nations World Population...  
1    Data sources: United Nations World Population ...  
2    Data sources: United Nations World Population ...  
3                   Estimates are based on regression.  
4    Data sources : United Nations World Population...  
..                                                 ...  
608  Data sources : Statistics South Africa, United...  
609  Data sources: United Nations World Population ...  
610  Data sources : United Nations World Population...  
611  Data sources : United Nations World Population...  
612  Data sources: United Nations World Population ...  

[613 rows x 3 columns]

#### 4 - <u>EdStatsSeries</u> <a id='cleaning-stats-series'></a>

In [206]:
# Display columns names
series.columns.tolist()

['Series Code',
 'Topic',
 'Indicator Name',
 'Short definition',
 'Long definition',
 'Unit of measure',
 'Periodicity',
 'Base Period',
 'Other notes',
 'Aggregation method',
 'Limitations and exceptions',
 'Notes from original source',
 'General comments',
 'Source',
 'Statistical concept and methodology',
 'Development relevance',
 'Related source links',
 'Other web links',
 'Related indicators',
 'License Type',
 'Unnamed: 20']

In [207]:
series = series[['Series Code','Topic','Indicator Name','Notes from original source','Other web links','Related indicators','License Type']]
series

Series Code         Topic  \
0             BAR.NOED.1519.FE.ZS    Attainment   
1                BAR.NOED.1519.ZS    Attainment   
2             BAR.NOED.15UP.FE.ZS    Attainment   
3                BAR.NOED.15UP.ZS    Attainment   
4             BAR.NOED.2024.FE.ZS    Attainment   
...                           ...           ...   
3660    UIS.XUNIT.USCONST.3.FSGOV  Expenditures   
3661    UIS.XUNIT.USCONST.4.FSGOV  Expenditures   
3662   UIS.XUNIT.USCONST.56.FSGOV  Expenditures   
3663  XGDP.23.FSGOV.FDINSTADM.FFD  Expenditures   
3664  XGDP.56.FSGOV.FDINSTADM.FFD  Expenditures   

                                         Indicator Name  \
0     Barro-Lee: Percentage of female population age...   
1     Barro-Lee: Percentage of population age 15-19 ...   
2     Barro-Lee: Percentage of female population age...   
3     Barro-Lee: Percentage of population age 15+ wi...   
4     Barro-Lee: Percentage of female population age...   
...                                                 ...   
3660  Government expenditure per upper secondary stu...   
3661  Government expenditure per post-secondary non-...   
3662  Government expenditure per tertiary student (c...   
3663  Government expenditure in secondary institutio...   
3664  Government expenditure in tertiary institution...   

      Notes from original source  Other web links  Related indicators  \
0                            NaN              NaN                 NaN   
1                            NaN              NaN                 NaN   
2                            NaN              NaN                 NaN   
3                            NaN              NaN                 NaN   
4                            NaN              NaN                 NaN   
...                          ...              ...                 ...   
3660                         NaN              NaN                 NaN   
3661                         NaN              NaN                 NaN   
3662                         NaN              NaN                 NaN   
3663                         NaN              NaN                 NaN   
3664                         NaN              NaN                 NaN   

      License Type  
0              NaN  
1              NaN  
2              NaN  
3              NaN  
4              NaN  
...            ...  
3660           NaN  
3661           NaN  
3662           NaN  
3663           NaN  
3664           NaN  

[3665 rows x 7 columns]

In [208]:
missing_values_table(series)

Displaying all columns with missing data: 



Missing Values  Missing Percentages (%)
Notes from original source            3665                    100.0
Other web links                       3665                    100.0
Related indicators                    3665                    100.0
License Type                          3665                    100.0

In [209]:
# Drop columns that contain +100% missing values
drop_columns(series, 100)

Displaying all columns with missing data: 

4 columns dropped because it they had 100% missing values: Notes from original source, Other web links, Related indicators, License Type


In [210]:
# Displaying the table with missing data filtered by percentage >= 100%
series

Series Code         Topic  \
0             BAR.NOED.1519.FE.ZS    Attainment   
1                BAR.NOED.1519.ZS    Attainment   
2             BAR.NOED.15UP.FE.ZS    Attainment   
3                BAR.NOED.15UP.ZS    Attainment   
4             BAR.NOED.2024.FE.ZS    Attainment   
...                           ...           ...   
3660    UIS.XUNIT.USCONST.3.FSGOV  Expenditures   
3661    UIS.XUNIT.USCONST.4.FSGOV  Expenditures   
3662   UIS.XUNIT.USCONST.56.FSGOV  Expenditures   
3663  XGDP.23.FSGOV.FDINSTADM.FFD  Expenditures   
3664  XGDP.56.FSGOV.FDINSTADM.FFD  Expenditures   

                                         Indicator Name  
0     Barro-Lee: Percentage of female population age...  
1     Barro-Lee: Percentage of population age 15-19 ...  
2     Barro-Lee: Percentage of female population age...  
3     Barro-Lee: Percentage of population age 15+ wi...  
4     Barro-Lee: Percentage of female population age...  
...                                                 ...  
3660  Government expenditure per upper secondary stu...  
3661  Government expenditure per post-secondary non-...  
3662  Government expenditure per tertiary student (c...  
3663  Government expenditure in secondary institutio...  
3664  Government expenditure in tertiary institution...  

[3665 rows x 3 columns]

#### 5 - <u>EdStatsFootNote</u> <a id='cleaning-stats-foot-note'></a>

In [211]:
# Display columns names
foot_note.columns.tolist()

['CountryCode', 'SeriesCode', 'Year', 'DESCRIPTION', 'Unnamed: 4']

In [212]:
missing_values_table(foot_note)

Displaying all columns with missing data: 



Missing Values  Missing Percentages (%)
Unnamed: 4          643638                    100.0

In [213]:
# Drop columns that contain +100% missing values
drop_columns(foot_note, 100)

Displaying all columns with missing data: 

1 column dropped because it had 100% missing values: Unnamed: 4


In [214]:
# Displaying the table with missing data filtered by percentage >= 100%
foot_note

CountryCode      SeriesCode    Year  \
0              ABW  SE.PRE.ENRL.FE  YR2001   
1              ABW  SE.TER.TCHR.FE  YR2005   
2              ABW  SE.PRE.TCHR.FE  YR2000   
3              ABW  SE.SEC.ENRL.GC  YR2004   
4              ABW     SE.PRE.TCHR  YR2006   
...            ...             ...     ...   
643633         ZWE     SH.DYN.MORT  YR2007   
643634         ZWE     SH.DYN.MORT  YR2014   
643635         ZWE     SH.DYN.MORT  YR2015   
643636         ZWE     SH.DYN.MORT  YR2017   
643637         ZWE     SP.POP.GROW  YR2017   

                                   DESCRIPTION  
0                          Country estimation.  
1                          Country estimation.  
2                          Country estimation.  
3                          Country estimation.  
4                          Country estimation.  
...                                        ...  
643633       Uncertainty bound is 91.6 - 109.3  
643634          Uncertainty bound is 54.3 - 76  
643635        Uncertainty bound is 48.3 - 73.3  
643636  5-year average value between 0s and 5s  
643637  5-year average value between 0s and 5s  

[643638 rows x 4 columns]

[Back to top](#top)